libs/platform/offline/src/services/sync-scheduler-default.service.ts

```diff
@@ -15,12 +15,17 @@ import { SyncSchedulerService } from './sync-scheduler.service';
 
 export const ProcessSyncsBackgroundSyncTag = 'oryx.ProcessSyncs';
 
+declare let self: ServiceWorkerGlobalScope;
+
 export class SyncSchedulerDefaultService implements SyncSchedulerService {
   protected scheduleSyncTimer?: ReturnType<typeof setTimeout>;
 
   constructor(
     protected indexedDbService = inject(IndexedDbService),
-    protected serviceWorker = navigator.serviceWorker
+    // navigator.serviceWorker is available only in main thread and this service is running in SW thread
+    protected serviceWorker:
+      | ServiceWorkerContainer
+      | undefined = navigator.serviceWorker
```

Comment:
What about:

```
protected async getServiceWorker() {
    return navigator.serviceWorker ? await navigator.serviceWorker.ready : self.registration:
}

```

and then:

`const sync = (await this.getServiceWorker()).sync;`